tests/acceptance/30_custom_promise_types/bodies.py

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/python3
+#
+# Sample custom promise type, uses cfengine.py library located in same dir.
+#
+# Use it in the policy like this:
+# promise agent append
+# {
+#   interpreter => "/usr/bin/python3";
+#   path => "$(sys.inputdir)/bodies.py";
+# }
+# body example_type example_name
+# {
+#   foo => "Hello";
+#   bar => "World";
+# }
+# bundle agent main
+# {
+#   append:
+#       "/path/to/target/file"
+#         example_type => example_name;
+# }
+
+
+import json
+from cfengine import PromiseModule, ValidationError, Result
+
+
+class BodiesPromiseTypeModule(PromiseModule):
+    def __init__(self):
+        super().__init__("bodies_promise_module", "0.0.1")
+
+    def validate_promise(self, promiser, attributes):
+        pass
+
+    def evaluate_promise(self, promiser, attributes):
+        try:
+            with open(promiser, "r") as f:
+                if (json.dumps(attributes) == f.read()):
+                    self.log_verbose("Promise '%s' kept" % promiser)
+                    return Result.KEPT
+                f.write(json.dumps(attributes))
```

Comment:
Does this `write()` on a file open for **reading** work? I guess it does, but please open the file for both reading and writing and also do `f.seek(0)` before the `write()` because otherwise it appends to the file (after what was read before).